src/Callback.php

```
@@ -139,8 +144,13 @@ public function isValid($value, $context = null)
                 $this->error(self::INVALID_VALUE);
                 return false;
             }
-        } catch (Exception) {
+        } catch (Throwable $error) {
```

Careful: `Throwable` used to be thrown by default :thinking:

---

Ack. I've pushed docs for the option - will this be sufficent do you think?

cc @froschdesign

---

Should validator be swallowing php errors?

---

I'd say not, but it's what we have

---

We don't have it. Yet. 
This change will make validator swallow php type errors, calling with fewer parameters or invoking callback that is not callable and such. All the subclasses of Error that I can think of are a direct result of a bug in the code.

---

Ah, I see what you mean.

Is this still true considering that `strict_types` is off here?

I think it's probably worth reverting this change anyhow. Ideally, nothing would be caught at all and this whole patch would be unnecessary.

---

I think this breakage is fine BTW, just requires some sort of release notes.

---

I've reverted back to `Exception`, you never know, someone out there might be catching `Error` 🤪

With your blessing @Xerkus I'll merge and release.

---

> Is this still true considering that `strict_types` is off here?

Huh. Unexpected.

Anyway, that leaves catchable errors coming from callback call stack.
I would add inline comment for later however. Something like "Intentionally excluding catchable errors as they are indicative of a bug and should not be suppressed"

My reasoning for allowing errors through is mostly because of user experience. Silly non-fatal mistake in the code can lead to confusing validation failure of a form or when tinkering with input filters. Debugging input filter specification can be daunting even without the error getting suppressed.